### ISSUE TITLE
[Bug][Pythonsrc] Incorrect tainted variable with tupleLiteral operator

### ISSUE THREAD

**Issue by prabhu** [joernio/joern#2663]
**Describe the bug**

Clone
https://github.com/HooliCorp/flask-webgoat

```
def source = cpg.call.code("request\\..*").inAssignment.argument.isIdentifier.filterNot(_.name.startsWith("tmp"))
def sink = cpg.call.code(".*query_db.*").argument.order(1)
sink.reachableByFlows(source).map(m => (m, m.elements.location.l)).toJsonPretty
```

When the above json is read and convert into a simplified tree, we can see the variable `username` is highlighted incorrectly instead of password.

<img width="1399" alt="2023-05-07_19-28-57" src="https://user-images.githubusercontent.com/7842/236693071-1fcac5d6-92da-41f2-9367-d776519c8e6c.png">

This is because the variable username is flagged as the symbol in the flow followed by the `tupleLiteral` operator.

```
{
        "packageName":"",
        "symbol":"username",
        "filename":"/home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py",
        "className":"",
        "methodFullName":"flask_webgoat/users.py:<module>.create_user",
        "lineNumber":39,
        "classShortName":"",
        "methodShortName":"create_user",
        "node":{
          "dynamicTypeHintFullName":[
            
          ],
          "name":"username",
          "code":"username",
          "typeFullName":"ANY",
          "columnNumber":12,
          "order":1,
          "_label":"IDENTIFIER",
          "argumentIndex":1,
          "lineNumber":39,
          "id":1673
        },
        "_label":"LOCATION",
        "nodeLabel":"IDENTIFIER"
      },
      {
        "packageName":"",
        "symbol":"(username, password, int(access_level))",
        "filename":"/home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py",
        "className":"",
        "methodFullName":"flask_webgoat/users.py:<module>.create_user",
        "lineNumber":39,
        "classShortName":"",
        "methodShortName":"create_user",
        "node":{
          "name":"<operator>.tupleLiteral",
          "signature":"",
          "code":"(username, password, int(access_level))",
          "typeFullName":"ANY",
          "columnNumber":11,
          "order":2,
          "methodFullName":"<operator>.tupleLiteral",
          "_label":"CALL",
          "argumentIndex":2,
          "dynamicTypeHintFullName":[
            
          ],
          "dispatchType":"STATIC_DISPATCH",
          "lineNumber":39,
          "id":1678
        },
        "_label":"LOCATION",
        "nodeLabel":"CALL"
      },
```

The problem is masked by joern's print mode since it doesn't highlight any specific symbol.

```
joern> sink.reachableByFlows(source).p
res247: List[String] = List(
  """________________________________________________________________________________________________________________________________________________________________________________
| nodeType   | tracked                                                          | lineNumber| method      | file                                                                |
|===============================================================================================================================================================================|
| Identifier | password = tmp3 = request.form tmp3.get("password")              | 20        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | password is None                                                 | 22        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | len(password)                                                    | 31        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | (username, password, int(access_level))                          | 39        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | (username, password, int(access_level))                          | 39        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Call       | (username, password, int(access_level))                          | 39        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Call       | "INSERT INTO user (username, password, access_level) VALUES (... | 38        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | query = "INSERT INTO user (username, password, access_level) ... | 37        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
| Identifier | query_db(query, [], False, True)                                 | 43        | create_user | /home/almalinux/work/HooliCorp/flask-webgoat/flask_webgoat/users.py |
""",
```

**Expected behavior**
Check tupleLiteral method

**Screenshots**

**Desktop (please complete the following information):**
 - OS
 - Joern Version 
 - Java version

**Additional context**

**Comment by DavidBakerEffendi** [comment]
Confirmed the bug here is that `tupleLiteral` does not have defined semantics. It is pretty tricky since this is a varargs-style call, but [this PR](https://github.com/joernio/joern/pull/2677) should pave the way for something like this. Specifically [here I have created a special mapping](https://github.com/joernio/joern/pull/2677/files#diff-31f555cf690908f81de5756bfe653161a81e0caa72426105a6a5c815fb44e013R135) that should essentially tell the engine that this call's arguments self-taint and also propagate to the return value.
